docs/PerformanceTest.md

```diff
@@ -0,0 +1,142 @@
+# eBPF Performance Benchmark Design
+
+## Overview
+
+The **eBPF Performance Benchmark suite (Perf)** aims to evaluate the performance of the eBPF runtime, pinpoint areas for enhancement, and ensure timely detection and mitigation of performance regressions. The suite provides a comprehensive view of build-to-build performance, categorized by scenarios, aiding in the identification of areas requiring improvement. Measuring the performance of the user-mode libbpf library is a non-goal as libbpf operations are assumed to occur at a lower frequency.
+
+## Architecture
+
+The Perf suite comprises these primary components:
+
+1. **Test Runner**: An executable designed to load, execute, and record metrics for an eBPF program.
+2. **Test Suites**: A collection of eBPF programs that comprehensively test various aspects of the eBPF runtime.
+3. **CI/CD Workflow**: A set of YAML files and scripts responsible for invoking the Test Runner with Test Suites, uploading resulting data to a central repository, and creating issues in case of performance regressions.
+
+## Test Runner
+
+The Test Runner (Runner) is a C++ program utilizing the libbpf library. It loads an eBPF program (in the kernel) and employs bpf_prog_test_run_opts to execute the BPF program on specified CPUs for a specified number of iterations. Subsequently, it calculates and outputs the average duration of the BPF programs. As a stretch goal, the Runner is designed to be platform-agnostic, enabling it to target both Windows and Linux eBPF runtimes. Given the importance of measuring both single threaded performance as well as concurrent performance, the test schema permit scheduling BPF programs on specific sets of CPUs.
+
+## Test Suites
+
+Each test is defined by a YAML file outlining test parameters and one or more associated eBPF programs. The parameters encompass:
+
+1. **Program -> CPU Assignment**: Specifies which CPUs execute each program, allowing concurrent execution.
```

Comment:
This list doesn't match the example YAML, which has name, description, elf_file, entry_point, and map_state_preparation.